sensors/triggers/params.go

```diff
@@ -203,52 +224,51 @@ func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*
 	case src.Value != nil:
 		// Use the default value set by the user in case the event is missing
 		resultValue = *src.Value
-		return &resultValue, nil
+		return &resultValue, stringType, nil
 	default:
 		// The parameter doesn't have a default value and is referencing a dependency that is
 		// missing in the received events. This is not an error and may happen with || conditions.
-		return nil, nil
+		return nil, stringType, nil
 	}
-
 	// If the event payload parsing failed
 	if err != nil {
 		// Fall back to the default value in case it exists
 		if src.Value != nil {
 			fmt.Printf("failed to parse the event payload, using default value. err: %+v\n", err)
 			resultValue = *src.Value
-			return &resultValue, nil
+			return &resultValue, stringType, nil
 		}
-
 		// Otherwise, return the error
-		return nil, err
+		return nil, errorType, err
 	}
-
 	// Get the value corresponding to specified key or template within event payload
 	if eventPayload != nil {
 		if tmplt != "" {
 			resultValue, err = getValueWithTemplate(eventPayload, tmplt)
 			if err == nil {
-				return &resultValue, nil
+				return &resultValue, "", nil
```

Comment:
What is the reason it returns `""` as the type instead of `stringType` when `err == nil`?

Reply:
done